collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/FormMapViewModel.java

```diff
@@ -66,129 +46,126 @@ public String getFormId() {
         return form.getJrFormId();
     }
 
-    public int getInstanceCount() {
+    /**
+     * Returns the count of all filled instances of this form, including unmappable ones.
+     */
+    public int getTotalInstanceCount() {
+        initializeFormInstances();
         return totalInstanceCount;
     }
 
-    public int getMappedPointCount() {
-        return points.size();
-    }
-
     /**
-     * Clears the existing features on the given map and places features for the current form's
-     * instances. If features were added to the map and the map view was not previously initialized,
-     * zooms to display all points.
+     * Returns a list of filled instances of this form that can be mapped.
      */
-    public void mapUpdateRequested(MapFragment map) {
-        points.clear();
-        map.clearFeatures();
+    public List<MappableFormInstance> getMappableFormInstances() {
+        initializeFormInstances();
+        return mappableFormInstances;
+    }
 
+    private void initializeFormInstances() {
         List<Instance> instances = instancesRepository.getAllBy(form.getJrFormId());
-        totalInstanceCount = instances.size();
-        for (Instance instance : instances) {
+
+        // Note: there is currently no way to delete instances from FormMapActivity so this works
+        // because a change in size means a refresh is needed. Compromise because we don't currently
+        // have an easy way to observe database changes.
+        if (mappableFormInstances == null || instances.size() != totalInstanceCount) {
```

Comment:
Kind of shady. Any alternative option I looked into -- getting some kind of callback when there's a DB change -- seemed terribly complex for not much gain. @seadowg you may have some ideas.

Reply:
I'm probable being dumb but I can work out how you end up with the instance number changing during the lifecycle of this View Model. Does the user not have to leave the map view to delete/add something and then when they come back they'll have a new View Model?

Reply:
This surprised me but it appears that the `ViewModel` is retained when the user goes to the form entry activity. I've done some digging just now and am even more confused. Do you have a reference that describes what you think should happen?

Reply:
Ah sorry you can click on an instance and then come back to this page right? That makes sense. The ViewModel will be retained while the Activity is in the back stack.

Reply:
Do you have a reference for this? I couldn't find it anywhere. The `Activity` itself gets killed, right?

Reply:
After chatting on Slack I think @lognaturel is right. We can get back to this screen after changing the instances so do need to think about reloads!

Reply:
What I was confused about was just the normal behavior for activities on the back stack. [They live on in a paused state](https://developer.android.com/guide/components/activities/activity-lifecycle) so all this makes sense. I'm not sure what I was getting confused about. Perhaps too much usage of "Don't keep activities."

What @seadowg was unsure about is the flow back to the map because typically exiting `FormEntryActivity` ignores the back stack and goes straight back to the main activity. We made a decision to return to the map on save because we want to enable a map-centric workflow where an enumerator uses the map as their landing place to fill and edit forms.

So I think we're all on the same page, now!